lang/typescript/src/extended-address/splitAddress1.ts

```diff
@@ -1,25 +1,52 @@
 import type {Address} from '../types/address';
-import {splitAddressField} from '../utils/address-fields';
-import {getRegionConfig, getConcatenationRules} from '../utils/regions';
+import {
+  RESERVED_DELIMITER,
+  splitAddressField,
+  regexSplitAddressField,
+} from '../utils/address-fields';
+import {
+  getRegionConfig,
+  getConcatenationRules,
+  getAddress1Regex,
+} from '../utils/regions';
 
 /**
- * Parse a concatenated address1 string based on the region specified by
- * country code
+ * Splits an address string into sub-fields using a reserved delimiter.
+ * Optionally provides a fallback mechanism to parse the address using
+ * a regex when the delimiter is absent, which should be used with caution
+ * as it may not provide accurate results.
+ *
  * @param countryCode 2-letter country code string
- * @param concatenatedAddress Combined address1 string
+ * @param address Combined address1 string
+ * @param experimentalRegexFallback Flag to attempt regex parsing as a fallback mechanism
  * @returns Partial Address object containing parsed address fields or null if
  * the region does not define an extended address format
  */
 export function splitAddress1(
   countryCode: string,
-  concatenatedAddress: string,
+  address: string,
+  experimentalRegexFallback = false,
 ): Partial<Address> | null {
   const config = getRegionConfig(countryCode);
   const fieldConcatenationRules = config
-    ? getConcatenationRules(config, concatenatedAddress, 'address1')
+    ? getConcatenationRules(config, address, 'address1')
     : undefined;
+  const address1Regex = config ? getAddress1Regex(config) : undefined;
+  if (address === '') {
+    return {};
+  }
```

Comment:
This will change existing behavior for parsing on countries that do not have a config thats untested right now. I'll suggest a code edit to include a test case to catch this behavior.

```ts
splitAddress1('US', '')) // returns null today
splitAddress1('US', '')) // returns {} with this change
```

We might want to invert this big if statement below and just return early:

```ts
if(!fieldConcatenationRules) {
  return null;
}

if (address === '') {
  return {};
}

// Now do the rest of the important config checks not in an if block
```

Reply:
Ah thanks for catching that edge case!